### PR TITLE
Allow `env_options` in ``Jinja2Templates`

### LINKED ISSUE #427: templateing: jinja2: pass kwargs for environment
I think it would be good to pass something like `env_kwargs` via https://github.com/blueyed/starlette/blob/24c135de71ac56a73f7f797258115941579155bf/starlette/templating.py#L51-L53.

While you can change the env afterwards, it would allow Jinja2 to validate e.g. `enable_async`, and call `load_extensions` etc.

### PR DESCRIPTION
**Pull request by aminalaee**
Closes #427 

Allows `Jinja2Templates` to set Jinja2 `Environment` options [here](https://jinja.palletsprojects.com/en/3.0.x/api/#jinja2.Environment).

For example:

```python
templates = Jinja2Templates(
        directory=directory, autoescape=False, auto_reload=True, enable_async=True
)
```

Or:

```python
loader = jinja2.FileSystemLoader(directory)

templates = Jinja2Templates(
        directory=directory, loader=loader
)
```

Since `directory` is still a required argument, it will be passed in this case, but the loader passed will take precedence over the `directory` and the loader created by Starlette. I can't think of any way to handle this right now, until maybe we can consider another API for Jinja2Templates later. This way we can deal with issues like #1214 .

- [x] Tests
- [x] Docs

### CODE REVIEW

File: starlette/templating.py [hunk 1]
```diff
@@ -55,20 +55,25 @@ class Jinja2Templates:
...
+        self, directory: typing.Union[str, PathLike], **env_options: typing.Any
     ) -> "jinja2.Environment":
         @pass_context
         def url_for(context: dict, name: str, **path_params: typing.Any) -> str:
             request = context["request"]
             return request.url_for(name, **path_params)
 
         loader = jinja2.FileSystemLoader(directory)
-        env = jinja2.Environment(loader=loader, autoescape=True)
+        env_options.setdefault("loader", loader)
```
Comment (aminalaee) [inline starlette/templating.py:73]: We can decide if we want to allow loader to be passed in env_options or not. The point is if we add it it makes `directory` reduntant but allows user full control over the loader.

**Review by lovelydinosaur — APPROVED** [review]
Neat. 👍

**Comment by aminalaee** [comment]
The documentation was not done yet. Can you check again?

**Comment by lovelydinosaur** [comment]
Yup - I oughta have noted that. 😅
Docs look good.

**Pull request merged**